### PR TITLE
migration-engine: fix migrations for some UnexecutableStepChecks with multiSchema

### PR DESCRIPTION
**Pull request by eviefp**
Fix a couple more migrations for `multiSchema`, specifically for `AlterTable` (and `AlterColumn`).

### CODE REVIEW

File: migration-engine/connectors/sql-migration-connector/src/sql_destructive_change_checker/unexecutable_step_check.rs [hunk 1]
```diff
@@ -5,33 +5,52 @@ use super::{
...
+    MadeScalarFieldIntoArrayField {
+        table: String,
+        column: String,
+        namespace: Option<String>,
+    },
+    DropAndRecreateRequiredColumn {
+        table: String,
+        column: String,
+        namespace: Option<String>,
+    },
```
Comment (tomhoule) [inline migration-engine/connectors/sql-migration-connector/src/sql_destructive_change_checker/unexecutable_step_check.rs:32]: Could the data in these enums be a good candidate for the `Table` and `Column` structs you defined in the previous PR?

**Review by tomhoule — APPROVED** [review]
We can do it later, but we'll have to test the destructive change checks in multiSchema mode. LGTM :+1:

**Comment by eviefp** [comment]
> We can do it later, but we'll have to test the destructive change checks in multiSchema mode. LGTM +1

Do you mean the others except the one that I already tested? Yeah, I might just do it this PR.

**Comment by tomhoule** [comment]
> Do you mean the others except the one that I already tested? Yeah, I might just do it this PR.

Ah you're right, the drop and recreate check code _is_ exercised and doesn't return any warnings because there is no data in the table. I was thinking about tests that do show destructive change warnings — I am not completely sure we need to, but we may also want to test that we include the namespaces in the warning messages when relevant.

**Pull request merged**